framework/include/csg/CSGBase.h

```diff
@@ -275,6 +291,301 @@ class CSGBase
     return _universe_list.getUniverse(name);
   }
 
+  /**
+   * @brief Create a 2D Cartesian Lattice from the provided universes. Each row in the lattice
+   * should be the same length. This version does not set an outer fill type (assumes VOID).
+   *
+   * @param name unique name identifier for the lattice
+   * @param pitch flat-to-flat size of the lattice elements
+   * @param universes list of list of universes that define the layout of the lattice
+   *
+   * @return reference to new Cartesian lattice
+   */
+  const CSGCartesianLattice & createCartesianLattice(
+      const std::string & name,
+      const Real pitch,
+      std::vector<std::vector<std::reference_wrapper<const CSGUniverse>>> universes);
+
+  /**
+   * @brief Create a 2D Cartesian Lattice from the provided universes with an outer material fill.
+   * Each row in the lattice should be the same length.
+   *
+   * @param name unique name identifier for the lattice
+   * @param pitch flat-to-flat size of the lattice elements
+   * @param universes list of list of universes that define the layout of the lattice
+   * @param outer_name name of material to use as outer fill between lattice elements
+   *
+   * @return reference to new Cartesian lattice
+   */
+  const CSGCartesianLattice & createCartesianLattice(
+      const std::string & name,
+      const Real pitch,
+      std::vector<std::vector<std::reference_wrapper<const CSGUniverse>>> universes,
+      const std::string & outer_name);
+
+  /**
+   * @brief Create a 2D Cartesian Lattice from the provided universes with an outer universe fill.
+   * Each row in the lattice should be the same length.
+   *
+   * @param name unique name identifier for the lattice
+   * @param pitch flat-to-flat size of the lattice elements
+   * @param universes list of list of universes that define the layout of the lattice
+   * @param outer_univ universe to use as outer fill between lattice elements
+   *
+   * @return reference to new Cartesian lattice
+   */
+  const CSGCartesianLattice & createCartesianLattice(
+      const std::string & name,
+      const Real pitch,
+      std::vector<std::vector<std::reference_wrapper<const CSGUniverse>>> universes,
+      const CSGUniverse & outer_univ);
+
+  /**
+   * @brief Create an empty 2D Cartesian lattice. This version does not set an outer fill type
+   * (assumes VOID).
+   * NOTE: this will not create any universe mapping; this must be done by using
+   * setLatticeUniverses.
```

Comment:
this simple routine ends up being duplicated too many times
3x for the different fills, 2x for whether universes are provided or not, 2x for the lattice type